internal/lib/advl4.go

```diff
@@ -44,3 +49,35 @@ func SetAdvL4Informers(c *AdvL4Informers) {
 func GetAdvL4Informers() *AdvL4Informers {
 	return AKOAdvL4Informers
 }
+
+func RemoveGatewayFinalizer(gw *advl4v1alpha1pre1.Gateway) {
+	finalizers := utils.Remove(gw.GetFinalizers(), GatewayFinalizer)
+	gw.SetFinalizers(finalizers)
+	UpdateGatewayFinalizer(gw)
+}
+
+func CheckAndSetGatewayFinalizer(gw *advl4v1alpha1pre1.Gateway) {
+	finalizers := gw.GetFinalizers()
+	if !ContainsFinalizer(gw, GatewayFinalizer) {
+		finalizers = append(finalizers, GatewayFinalizer)
+	}
+
+	gw.SetFinalizers(finalizers)
```

Comment:
Looks like this function can be optimised. If you discover that the finalisers are already present on the gateway, there's no need to re-update it right?

Reply:
yes! thanks